### ISSUE TITLE
Add media remove admin API for remote media to remove it from disk

### ISSUE THREAD

**Issue by Zocker1999NET** [matrix-org/synapse#16072]
**Description:**

Given an user of mine receives media with illegal content and reports it to me, I need to be able to remove it from my server. Putting the file in quarantine is good, but for most illegal content not good enough, as it may still be illegal to keep such files.

So I request an API (or it may become an option of quarantine), which allows admins to remove such files, if identified from the server. Together with the kept entry for quarantine, following my tests, this effectively blocks my server from downloading the file again, making it inaccessible to my users & to the federation via my server.

This can be achieved by hand by logging in to the server, navigating to the media repository, checking the filesystem id of the remote media in the database, and then deleting the files. But that takes way more time & requires more privileges & might create inconsistencies for Synapse. A builtin API can make this process way faster while Synapse can (ideally) prevent & handle such inconsistencies.

**Comment by DMRobertson** [comment]
Does ["Delete a specific local media"](https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#delete-a-specific-local-media) suffice here?

**Comment by Zocker1999NET** [comment]
> Does ["Delete a specific local media"](https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#delete-a-specific-local-media) suffice here?

No. This seems for deleting local media (originally uploaded by users of my server), but I need to delete remote media (originally uploaded by users of other servers). And if I try deleting the media with this API, Synapse answers with: `Can only delete local media`.

**Comment by DMRobertson** [comment]
In that case, https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#purge-remote-media-api ("Purge remote media") sounds like what you want.

**Comment by DMRobertson** [comment]
Oh sorry, I should have read what that endpoint does more carefully. I'm surprised we don't have something for deleting a specific piece of remote media as you say. Thanks for flagging this up.
